rio_tiler_pds/landsat/aws/landsat_collection2.py

```diff
@@ -0,0 +1,166 @@
+    "QA_PIXEL",
+    "QA_RADSAT",
+    "SR_B1",
+    "SR_B2",
+    "SR_B3",
+    "SR_B4",
+    "SR_B5",
+    "SR_B6",
+    "SR_B7",
+    "SR_QA_AEROSOL",
+)
+
+OLI_TIRS_ST_BANDS = (
+    "ST_ATRAN",
+    "ST_B10",
+    "ST_CDIST",
+    "ST_DRAD",
+    "ST_EMIS",
+    "ST_EMSD",
+    "ST_QA",
+    "ST_TRAD",
+    "ST_URAD",
+)
+
+# For collection 2, level 2, ETM and TM have the same bands
+TM_SR_BANDS = (
+    "QA_PIXEL",
+    "QA_RADSAT",
+    "SR_ATMOS_OPACITY",
+    "SR_B1",
+    "SR_B2",
+    "SR_B3",
+    "SR_B4",
+    "SR_B5",
+    "SR_B7",
+    "SR_CLOUD_QA",
+)
+
+TM_ST_BANDS = (
+    "QA_PIXEL",
+    "QA_RADSAT",
+    "ST_ATRAN",
+    "ST_B6",
+    "ST_CDIST",
+    "ST_DRAD",
+    "ST_EMIS",
+    "ST_EMSD",
+    "ST_QA",
+    "ST_TRAD",
+    "ST_URAD",
+)
+
+
+@attr.s
+class LandsatC2L2Reader(MultiBandReader):
+    """AWS Public Dataset Landsat Collection 2 Level 2 COG Reader.
+
+    Args:
+        sceneid (str): Landsat 8 sceneid.
+
+    Attributes:
+        minzoom (int): Dataset's Min Zoom level (default is 5).
+        maxzoom (int): Dataset's Max Zoom level (default is 12).
+        scene_params (dict): scene id parameters.
+        bands (tuple): list of available bands.
+
+    Examples:
+        >>> with L8C2COGReader('LC08_L2SR_093106_20200207_20201016_02_T2') as scene:
+                print(scene.bounds)
+
+        >>> with L8C2COGReader('LC08_L1TP_116043_20201122_20201122_02_RT') as scene:
+                print(scene.bounds)
+
+    """
+
+    sceneid: str = attr.ib()
+    reader: Type[COGReader] = attr.ib(default=COGReader)
+    reader_options: Dict = attr.ib(default={})
+    tms: TileMatrixSet = attr.ib(default=WEB_MERCATOR_TMS)
+    minzoom: int = attr.ib(default=5)
+    maxzoom: int = attr.ib(default=12)
+
+    bands: Tuple = attr.ib(init=False)
+
+    _scheme: str = "s3"
+    _hostname: str = "usgs-landsat"
+    _prefix: str = "collection02/level-{_processingLevelNum}/standard/{_sensor}/{acquisitionYear}/{path}/{row}/{scene}/{scene}"
+
+    def __attrs_post_init__(self):
+        """Fetch productInfo and get bounds."""
+        self.scene_params = sceneid_parser(self.sceneid)
+
+        processing_level = self.scene_params["processingCorrectionLevel"]
+        sensor = self.scene_params["_sensor"]
+
+        # For collection 2, level 2, ETM and TM have the same bands
+        if processing_level == "L2SR":
+            if sensor == "oli-tirs":
+                self.bands = OLI_TIRS_SR_BANDS
+            elif sensor == "tm":
+                self.bands = TM_SR_BANDS
+            elif sensor == "etm":
+                self.bands = TM_SR_BANDS
+        else:
+            if sensor == "oli-tirs":
+                self.bands = OLI_TIRS_SR_BANDS + OLI_TIRS_ST_BANDS
+            elif sensor == "tm":
+                self.bands = TM_SR_BANDS + TM_ST_BANDS
+            elif sensor == "etm":
+                self.bands = TM_SR_BANDS + TM_ST_BANDS
+        # TODO: add separate level 1 reader with TOA reflectances
```

Comment:
@kylebarron why do you think we need a separate reader for level 1 and level 2? IMO one reader can be enough. 

If you think it's because of TOA, I don't think we need to care about creating TOA values ... I'm pretty sure advanced users will:
- 1. wait for level 2 data to be created (it's only 15 to 27 days be processed from L1 to L2, https://www.usgs.gov/media/images/landsat-collection-2-generation-timeline)
- 2. download the data and do their own translation.

To create TOA from Level 1 we will need to fetch the MTL file (and maybe the ang file) which is not really optimized, so I'm happy to just say rio-tiler-pds doesn't support live transformation to TOA

Reply:
I originally was going to put both in one reader but you seemed to want to support TOA values for level 1. We could combine them into one and if someone wants TOA values in the future we could separate them then